juju/application.py

```diff
@@ -706,19 +701,23 @@ async def refresh(
         if parsed_url.schema is None:
             raise JujuError(f'A ch: or cs: schema is required for application refresh, given : {str(parsed_url)}')
 
-        if revision is not None:
-            origin.revision = revision
+        current_origin = charm_url_origin_result.charm_origin
```

Comment:
because of this move, the parse above might fail for the local urls. I believe that's the reason for [this failure](https://github.com/juju/python-libjuju/actions/runs/9778130375/job/26994335382?pr=1067#step:6:462).

Reply:
Good spot. The order of things seems to lack any real order here. I'll move things around a little